bxsolana/provider/ws.ts

```diff
@@ -62,233 +62,305 @@ import {
+
     constructor(address: string = MAINNET_API_WS) {
         super()
+        this.wsConnection = new RpcWsConnection(address)
         this.address = address
     }
 
+    async connect() {
+        await this.wsConnection.connect()
+    }
+
     close = () => {
         this.isClosed = true
-        this.socket.close()
+        this.wsConnection.close()
     }
 
-    getOrderbook = (request: GetOrderbookRequest): Promise<GetOrderbookResponse> => {
-        return this.wsSocketCall("GetOrderbook", request)
+    async getOrderbook(request: GetOrderbookRequest): Promise<GetOrderbookResponse> {
+        return await this.wsConnection.call("GetOrderbook", request)
     }
 
-    getMarkets = (request: GetMarketsRequest): Promise<GetMarketsResponse> => {
-        return this.wsSocketCall("GetMarkets", request)
+    async getMarkets(request: GetMarketsRequest): Promise<GetMarketsResponse> {
+        return await this.wsConnection.call("GetMarkets", request)
     }
 
-    getTickers(request: GetTickersRequest): Promise<GetTickersResponse> {
-        return this.wsSocketCall("GetTickers", request)
+    async getTickers(request: GetTickersRequest): Promise<GetTickersResponse> {
+        return await this.wsConnection.call("GetTickers", request)
     }
 
-    getTrades(request: GetTradesRequest): Promise<GetTradesResponse> {
-        return this.wsSocketCall("GetTrades", request)
+    async getTrades(request: GetTradesRequest): Promise<GetTradesResponse> {
+        return await this.wsConnection.call("GetTrades", request)
     }
 
-    getServerTime(request: GetServerTimeRequest): Promise<GetServerTimeResponse> {
-        return this.wsSocketCall("GetServerTime", request)
+    async getServerTime(request: GetServerTimeRequest): Promise<GetServerTimeResponse> {
+        return await this.wsConnection.call("GetServerTime", request)
     }
 
-    getOpenOrders(request: GetOpenOrdersRequest): Promise<GetOpenOrdersResponse> {
-        return this.wsSocketCall("GetOpenOrders", request)
+    async getOpenOrders(request: GetOpenOrdersRequest): Promise<GetOpenOrdersResponse> {
+        return await this.wsConnection.call("GetOpenOrders", request)
     }
 
-    getUnsettled(request: GetUnsettledRequest): Promise<GetUnsettledResponse> {
-        return this.wsSocketCall("GetUnsettled", request)
+    async getUnsettled(request: GetUnsettledRequest): Promise<GetUnsettledResponse> {
+        return await this.wsConnection.call("GetUnsettled", request)
     }
 
-    getAccountBalance(request: GetAccountBalanceRequest): Promise<GetAccountBalanceResponse> {
-        return this.wsSocketCall("GetAccountBalance", request)
+    async getAccountBalance(request: GetAccountBalanceRequest): Promise<GetAccountBalanceResponse> {
+        return await this.wsConnection.call("GetAccountBalance", request)
+    }
+
+    manageGetStreamMaps = async (streamName: string, subscriptionID: string) => {
+        let count: number
+        if (this.streamToCountMap.size === 0) {
```

Comment:
`if (!(streamName in this.streamToCountMap))`. Otherwise this only gets called once for all different streams